src/general/IWorldEffect.cs

```diff
@@ -33,17 +34,21 @@ public void OnRegisterToWorld()
 
     public void OnTimePassed(double elapsed, double totalTimePassed)
     {
+        var glucose = SimulationParameters.Instance.GetCompound("glucose");
+
         foreach (var key in targetWorld.Map.Patches.Keys)
         {
             var patch = targetWorld.Map.Patches[key];
 
-            foreach (var compound in patch.Biome.Compounds.Keys)
+            // If there are microbes to be eating up the primordial soup, reduce the milk
+            if (patch.SpeciesInPatch.Count > 0)
             {
-                if (compound.InternalName == "glucose")
+                EnvironmentalCompoundProperties glucoseValue;
+                if (patch.Biome.Compounds.TryGetValue(glucose, out glucoseValue))
```

Comment:
the variable declaration can be inlined in the out statement

Reply:
C# is weird.